extras/indexing/src/main/java/org/apache/rya/indexing/IndexPlanValidator/ValidIndexCombinationGenerator.java

```diff
@@ -86,11 +81,8 @@ public boolean hasNext() {
 
 					}
 
-				} else if (isEmpty) {
-					return false;
-				} else {
-					return true;
-				}
+				} else
+                    return !isEmpty;
```

Comment:
do not remove curly braces around the else

Reply:
This was my idea. Sorry